src/py/flwr/server/driver/grpc_driver.py

```diff
@@ -221,24 +227,58 @@ def push_messages(self, messages: Iterable[Message]) -> Iterable[str]:
                 "list has `None` for those messages (the order is preserved as passed "
                 "to `push_messages`). This could be due to a malformed message.",
             )
+        # Store message IDs
+        self._message_ids.extend(res.message_ids)
```

Comment:
If we switch to `set`, we need to use `update` here instead.

```suggestion
        self._message_ids.update(res.message_ids)
```